src/models/state/mod.rs

```diff
@@ -801,11 +826,9 @@ impl GlobalState {
         // loop over all monitored utxos
         let monitored_utxos = self.wallet_state.wallet_db.monitored_utxos_mut();
 
-        // note: iter_mut_lock holds a write-lock, so it should be dropped
-        // immediately after use.
-        let mut iter_mut_lock = monitored_utxos.iter_mut();
-        'outer: while let Some(mut setter) = iter_mut_lock.next() {
-            let monitored_utxo = setter.value();
+        'outer: for i in 0..monitored_utxos.len().await {
```

Comment:
Same as previous comment: looks like DB IO inside of a loop.

Reply:
In this case, both the previous code and new code are retrieving and setting elements one-by-one inside the loop.  They are equivalent.

The old code is using a mutating iterator which provides a setter, which holds the value and allows mutation.

Mutating streams do not exist afaik, so I was unable to duplicate the mutating iterator in the storage layer.  Instead, the new code uses regular get() and set() with indexes, which is logically the same.